compiler/src/main/java/com/xfinity/resourceprovider/RpCodeGenerator.java

```diff
@@ -50,38 +51,68 @@ TypeSpec generateResourceProviderClass() {
+                                           .addStatement("this.integerProvider = new IntegerProvider(context)")
                                            .build();
 
         ClassName stringProviderClassName = get(packageName, "StringProvider");
-        FieldSpec stringProvider = FieldSpec.builder(stringProviderClassName, "string")
-                                            .addModifiers(Modifier.PUBLIC, Modifier.FINAL).build();
+        FieldSpec stringProvider = FieldSpec.builder(stringProviderClassName, "stringProvider")
+                                            .addModifiers(Modifier.PRIVATE, Modifier.FINAL).build();
 
         ClassName drawableProviderClassName = get(packageName, "DrawableProvider");
-        FieldSpec drawableProvider = FieldSpec.builder(drawableProviderClassName, "drawable")
-                                              .addModifiers(Modifier.PUBLIC, Modifier.FINAL).build();
+        FieldSpec drawableProvider = FieldSpec.builder(drawableProviderClassName, "drawableProvider")
+                                              .addModifiers(Modifier.PRIVATE, Modifier.FINAL).build();
 
         ClassName colorProviderClassName = get(packageName, "ColorProvider");
-        FieldSpec colorProvider = FieldSpec.builder(colorProviderClassName, "color")
-                                           .addModifiers(Modifier.PUBLIC, Modifier.FINAL).build();
+        FieldSpec colorProvider = FieldSpec.builder(colorProviderClassName, "colorProvider")
+                                           .addModifiers(Modifier.PRIVATE, Modifier.FINAL).build();
 
         ClassName dimenProviderClassName = get(packageName, "DimensionProvider");
-        FieldSpec dimenProvider = FieldSpec.builder(dimenProviderClassName, "dimen")
-                                           .addModifiers(Modifier.PUBLIC, Modifier.FINAL).build();
+        FieldSpec dimenProvider = FieldSpec.builder(dimenProviderClassName, "dimenProvider")
+                                           .addModifiers(Modifier.PRIVATE, Modifier.FINAL).build();
 
         ClassName integerProviderClassName = get(packageName, "IntegerProvider");
-        FieldSpec integerProvider = FieldSpec.builder(integerProviderClassName, "integer")
-                                             .addModifiers(Modifier.PUBLIC, Modifier.FINAL).build();
-
+        FieldSpec integerProvider = FieldSpec.builder(integerProviderClassName, "integerProvider")
+                                             .addModifiers(Modifier.PRIVATE, Modifier.FINAL).build();
+
+        MethodSpec getStringMethodSpec = MethodSpec.methodBuilder("getString")
+                                                   .addModifiers(Modifier.PUBLIC)
+                                                   .addStatement("return stringProvider")
+                                                   .returns(get(packageName, "StringProvider"))
```

Comment:
The ClassName is defined for all of these previously, you could just reference those variables here.

Reply:
will do.

Reply:
Fixed